pipeline/pipeline/backend.py

```diff
@@ -181,48 +181,36 @@ def _run(self, pipeline, dry_run, verbose, delete_scratch_on_exit):  # pylint: d
         task_to_job_mapping = {}
         job_id_to_command = {}
 
-        volumes = [{'volume': {'name': 'pipeline-test-0-1--hail-is-service-account-key',
-                               'secret': {'optional': False,
-                                          'secretName': 'pipeline-test-0-1--hail-is-service-account-key'}},
-                    'volume_mount': {'mountPath': '/secrets',
-                                     'name': 'pipeline-test-0-1--hail-is-service-account-key',
-                                     'readOnly': True}}]
-
-        activate_service_account = 'gcloud auth activate-service-account ' \
-                                   'pipeline-test-0-1--hail-is@hail-vdc.iam.gserviceaccount.com ' \
-                                   '--key-file /secrets/pipeline-test-0-1--hail-is.key'
-
         for task in pipeline._tasks:
             def copy_input(r):
                 if isinstance(r, ResourceFile):
                     if isinstance(r, InputResourceFile):
-                        return [f'gsutil cp {r._input_path} {r._get_path(local_tmpdir)}']
+                        return [(r._input_path, r._get_path(local_tmpdir))]
                     else:
                         assert isinstance(r, TaskResourceFile)
-                        return [f'gsutil cp {r._get_path(remote_tmpdir)} {r._get_path(local_tmpdir)}']
+                        return [(r._get_path(remote_tmpdir), r._get_path(local_tmpdir))]
                 else:
                     assert isinstance(r, ResourceGroup)
                     return [x for _, rf in r._resources.items() for x in copy_input(rf)]
 
             def copy_output(r):
                 assert r._source == task  # pylint: disable-msg=W0640
                 if isinstance(r, TaskResourceFile):
-                    return [f'gsutil cp {r._get_path(local_tmpdir)} {r._get_path(remote_tmpdir)}']
+                    return [
+                        (r._get_path(local_tmpdir), r._get_path(remote_tmpdir)),
+                        *[(r._get_path(local_tmpdir), output_path) for output_path in r._output_paths]]
```

Comment:
@akotlar Please give me a chance to review the changes to Pipeline before approving. Thanks!

Reply:
Will do

Reply:
@danking I think these changes should go in a separate PR. The issue is `r._output_paths` only has the task output files and not the input resource files that may also want to be written with `p.write_output`. Also, a task output file might be written to multiple places besides the temp bucket with `p.write_output`. Lastly, I handled writing out the entire resource group with the extensions (this logic will change in the future once I get the resource group interface nailed down). 

Are these changes to pipeline necessary for this PR? If not, I propose removing them and letting me fix it in a separate PR.

Reply:
I removed the changes. Now this PR only changes pipeline's gitignore (it is missing some files that my Emacs produces), changes necessary for the new authentication stuff, and I changed the image to be tag 237.0.0-alpine, which this PR makes the default google/cloud-sdk tag everywhere.